slam:
  ros__parameters:
    # Determines the mission type
    # (0:acceleration, 1:skidpad, 2:autoX, 3:trackdrive)
    mission_type: 0
    # Determines if mapping mode is on (false => localization only, no new cones added)
    mapping_mode: false

    # All parameters related to data association
    data_association:
      # Data association algorithm ("nearest_neighbor")
      algorithm: "nearest_neighbor"
      # Parameters related to nearest neighbor data association
      nearest_neighbor:
        # If a new cone is not within this distance from
        # an existing cone, it will be treated as a new cone in the graph.
        # Otherwise it will be associated to an existing cone.
        min_distance_threshold_m: 1.0

    # All parameters related to the slam backend
    slam_backend:
      # Slam backend algorithm ("graph_slam", "fixed_lag_smoothing", "isam2")
      algorithm: "graph_slam"
      # Parameters related to graph slam
      graph_slam:
        # The maximum allowed total edges in the graph
        max_edges: 10000
        # TODO(Christoforos) Future modifications for performance
        # Maximium number of iterations for the GTSAM optimizer
        optimizer_iterations: 3
        # Error tolerance for GTSAM optimizer
        optimizer_error_tolerance: 0.1
        # Noise model: prior of odometry
        pose_prior_noise: [0.01, 0.01, 0.01]
        # Noise model: odometry
        pose_noise: [0.1, 0.1, 0.1]
        # Noise model: prior of landmarks (fixed map)
        landmark_prior_noise: [1.0, 1.0]
        # Noise model: cone observations (bearing, range)
        landmark_noise: [0.2, 1.0]

    # GlobalMapTracker parameters
    global_map_tracker:
      probability_updates_enabled: false
      cone_ignore_enabled: false

    map_loader:
      map:
        # Parameters for the acceleration map
        acceleration:
          # The track length from start to finish line in meters
          track_first_part_length_m: 75.0
          # The track length from finish line until end of orange cones in meters
          track_second_part_length_m: 75.0
          # The number of blue or yellow cones in one of the boundaries (counted manually)
          # NOT INCLUDING ORANGE CONES!
          num_boundary_cones: 74 # 14
          # The number of orange cones after finish line in one of the boundaries (counted manually)
          # NOT INCLUDING BIG ORANGE CONES!
          num_boundary_orange_cones: 75 # 15
          # Max track width (blue to yellow) distance meters
          max_track_width_y_m: 7.0 # 6.0
          # Distance from car to first blue/yellow cones (estimated manually)
          # Cones less than this distance are ignored
          estimated_first_cones_distance_x_m: 3.0
          # Distance between big orange cones in x (estimated manually)
          estimated_big_orange_cone_distance_x_m: 1.0
          # Minimum distance between blue/yellow cones in x in meters
          min_cone_distance_x_m: 2.0
          # Maximum distance between blue/yellow cones in x in meters
          max_cone_distance_x_m: 5.5
          # How much weight is put on perception measured distance
          perception_observation_weight: 0.0 # 0.2
          # How many cones should be detected (per boundary) to build a good map
          min_num_detected_cones: 2
